manim/animation/composition.py

```diff
@@ -31,9 +31,15 @@ def __init__(
     ) -> None:
         self.animations = [prepare_animation(anim) for anim in animations]
         self.group = group
+        self.animations_with_mobjects = []
+        for anim in animations:
+            if isinstance(anim.mobject, Mobject):
+                self.animations_with_mobjects.append(anim)
         if self.group is None:
             self.group = Group(
-                *remove_list_redundancies([anim.mobject for anim in animations])
+                *remove_list_redundancies(
+                    [anim.mobject for anim in self.animations_with_mobjects]
+                )
```

Comment:
Would prefer to not keep an extra list of animations with `self.animations_with_mobjects` and would also prefer checking if `anim.mobject` is `None` instead because `Animation` errors if the mobject is not `None` or an instance of `Mobject`, and checking against `None` requires less overhead.
```suggestion
                *remove_list_redundancies(
                    [anim.mobject for anim in self.animations if anim.mobject is not None]
                )
```

Perhaps an extra method/property/attribute could be added to `Animation` like `is_dummy` that could be checked instead of doing this here.

Reply:
Thanks for the feedback! I forgot that you could use conditional logic in list comprehension. I've committed the change related to that, and I'm now playing around with writing an `is_dummy` method per your suggestion which could be used anywhere to determine if an animation has a child Mobject.